tutu-backend/src/main/scala/fi/oph/tutu/backend/service/perusteluMuistio/PerusteluMuistioGenerator.scala

```diff
@@ -186,20 +191,128 @@ def haeTutkintokohtaisetTiedot(
                 }
               )
 
+          val suoritusvuodet = Seq(tutkinto.aloitusVuosi, tutkinto.paattymisVuosi).flatten
+            .mkString(" - ")
+
           Seq[String](
             s"Tutkinto ${tutkinto.jarjestys}:",
             s"  Tutkintotodistusotsikko: ${tutkinto.todistusOtsikko.getOrElse("-")}",
             s"  Nimi: ${tutkinto.nimi.getOrElse("-")}",
             s"  Pääaine tai erikoisala: ${tutkinto.paaaaineTaiErikoisala.getOrElse("-")}",
             s"  Korkeakoulun tai oppilaitoksen nimi: ${tutkinto.oppilaitos.getOrElse("-")}",
             s"  Korkeakoulun tai oppilaitoksen sijaintimaa: ${kielistettyMaakoodi.getOrElse("-")}",
-            s"  Todistuksen päivämäärä: ${tutkinto.todistuksenPaivamaara.getOrElse("-")}"
+            s"  Todistuksen päivämäärä: ${tutkinto.todistuksenPaivamaara.getOrElse("-")}",
+            s"  Suoritusvuodet: ${suoritusvuodet}",
+            s"  Ohjeellinen laajuus: ${tutkinto.ohjeellinenLaajuus.getOrElse("-")}",
+            s"  Tutkintoon sisältyi opinnäytetyö: ${tutkinto.opinnaytetyo.map(toKyllaEi).getOrElse("-")}",
+            s"  Tutkintoon sisältyi harjoittelu: ${tutkinto.harjoittelu.map(toKyllaEi).getOrElse("-")}",
+            s"  Lisätietoja opinnäytteisiin tai harjoitteluun liittyen: ${tutkinto.perustelunLisatietoja.getOrElse("-")}"
           ).mkString("\n")
         })
         .mkString("\n\n")
     })
 }
 
+def haeYleisetPerustelut(perusteluMaybe: Option[Perustelu]): Option[String] = {
+  if (perusteluMaybe == None) {
+    None
+  } else {
+    val perustelu = perusteluMaybe.get
+
+    val resultString = Seq(
+      perustelu.virallinenTutkinnonMyontaja
+        .map(toKyllaEi)
+        .map(muotoiltuValue => s"Virallinen tutkinnon myöntäjä: ${muotoiltuValue}"),
+      perustelu.virallinenTutkinto
+        .map(toKyllaEi)
+        .map(muotoiltuValue => s"Virallinen tutkinto: ${muotoiltuValue}"),
+      if (perustelu.lahdeLahtomaanKansallinenLahde) {
+        Some("Lähde: Lähtömaan kansallinen lähde (verkkosivut, lainsäädäntö, julkaisut)")
+      } else None,
+      if (perustelu.lahdeLahtomaanVirallinenVastaus) {
+        Some("Lähde: Lähtömaan virallinen vastaus")
+      } else None,
+      if (perustelu.lahdeKansainvalinenHakuteosTaiVerkkosivusto) {
+        Some("Lähde: Kansainvälinen hakuteos tai verkkosivusto")
+      } else None,
+      if (perustelu.selvitysTutkinnonMyontajastaJaTutkinnonVirallisuudesta != "") {
+        Some(
+          s"Lyhyt selvitys tutkinnon myöntäjästä ja tutkinnon virallisuudesta:\n${perustelu.selvitysTutkinnonMyontajastaJaTutkinnonVirallisuudesta}"
+        )
+      } else None,
+      perustelu.ylimmanTutkinnonAsemaLahtomaanJarjestelmassa
+        .map(asema =>
+          asema match {
+            case "alempi_korkeakouluaste"           => "Vähintään kolmivuotinen ensimmäisen vaiheen korkeakoulututkinto"
```

Comment:
```suggestion
.map {
            case "alempi_korkeakouluaste" => "Vähintään kolmivuotinen ensimmäisen vaiheen korkeakoulututkinto"
```

Idea ehottelis tämmöstä syntaksia, toimiskohan?